api/v1alpha1/oadp_types.go

```diff
@@ -54,47 +54,81 @@ const CSIPluginImageKey UnsupportedImageKey = "csiPluginImageFqin"
-	// VeleroResourceAllocations defines the CPU and Memory resource allocations for the Velero Pod
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Tolerations defines the list of tolerations to be applied to Restic daemonset
 	// +optional
-	VeleroResourceAllocations corev1.ResourceRequirements `json:"veleroResourceAllocations,omitempty"`
-	// DefaultVeleroPlugins defines the list of default plugins (aws, gcp, azure, openshift, csi) to be installed with Velero
-	DefaultVeleroPlugins []DefaultPlugin `json:"defaultVeleroPlugins,omitempty"`
-	// CustomVeleroPlugins defines the custom plugin to be installed with Velero
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// ResourceAllocations defines the CPU and Memory resource allocations for the restic Pod
 	// +optional
-	CustomVeleroPlugins []CustomPlugin `json:"customVeleroPlugins,omitempty"`
+	ResourceAllocations corev1.ResourceRequirements `json:"resourceAllocations,omitempty"`
+}
 
-	// EnableRestic is a boolean to specify if restic daemonset instance should be created or not
+// ResticConfig is the configuration for restic server
+type ResticConfig struct {
+	// Enable defines a boolean pointer whether we want the daemonset to
+	// exist or not
 	// +optional
-	EnableRestic *bool `json:"enableRestic,omitempty"`
-	// ResticSupplementalGroups defines the linux groups to be applied to the Restic Pod
+	Enable *bool `json:"enable,omitempty"`
+	// SupplementalGroups defines the linux groups to be applied to the Restic Pod
 	// +optional
-	ResticSupplementalGroups []int64 `json:"resticSupplementalGroups,omitempty"`
-	// ResticNodeSelector defines the nodeSelector to be supplied to Restic podSpec
+	SupplementalGroups []int64 `json:"supplementalGroups,omitempty"`
+	// Timeout defines the Restic timeout, default value is 1h
 	// +optional
-	ResticNodeSelector map[string]string `json:"resticNodeSelector,omitempty"`
-	// ResticTolerations defines the list of tolerations to be applied to Restic daemonset
-	// +optional
-	ResticTolerations []corev1.Toleration `json:"resticTolerations,omitempty"`
-	// ResticResourceAllocations defines the CPU and Memory resource allocations for the restic Pod
+	Timeout string `json:"timeout,omitempty"`
+	// Pod specific configuration
+	PodConfig *PodConfig `json:"podConfig,omitempty"`
+}
+
+// ApplicationConfig defines the configuration for the Data Protection Application
+type ApplicationConfig struct {
+	Velero *VeleroConfig `json:"velero,omitempty"`
+	Restic *ResticConfig `json:"restic,omitempty"`
+}
+
+// BackupLocation defines the configuration for the DPA backup storage
+type BackupLocation struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Velero            *velero.BackupStorageLocationSpec `json:"velero"`
+}
+
+// SnapshotLocation defines the configuration for the DPA snapshot store
+type SnapshotLocation struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
is the entire surface of `metav1.ObjectMeta` required here? From what I recall we needed stuff like name, labelselector etc. I'm a little concerned about introducing the object meta here that we will not be able to support in the future.

Reply:
That's a fair point. My thought on this was to support labels/annotations. But you're right, we are opening the surface area pretty far.